File: src/1248_Count_Number_of_Nice_Subarrays/solution.py
## https://leetcode.com/problems/count-number-of-nice-subarrays/
from typing import List


## TC: O(N)
## SC: O(1)
def numberOfSubarrays(nums: List[int], k: int) -> int:
    count_odds = 0
    count_nice = 0
    ans = 0
    l = 0
    for num in nums:
        if num % 2 == 1:
            count_odds += 1
            count_nice = 0
        while count_odds == k:
            if nums[l] % 2 == 1:
                count_odds -= 1
            count_nice += 1
            l += 1
        ans += count_nice
    return ans

File: src/1248_Count_Number_of_Nice_Subarrays/solution.ts
// https://leetcode.com/problems/count-number-of-nice-subarrays/

// TC: O(N)
// SC: O(1)
function numberOfSubarrays(nums: number[], k: number): number {
  let l = 0;
  let countOdds = 0;
  let countNice = 0;
  let ans = 0;
  for (let r = 0; r < nums.length; r++) {
    if (nums[r] % 2 == 1) {
      countOdds += 1;
      countNice = 0;
    }
    while (countOdds == k) {
      if (nums[l] % 2 == 1) {
        countOdds -= 1;
      }
      l += 1;
      countNice += 1;
    }
    ans += countNice;
  }
  return ans;
}
